master/usage/policy/secure-metrics.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Securing access to {{site.prodname}}'s metrics endpoints
+---
+
+## About securing access to {{site.prodname}}'s metrics endpoints
+
+When using {{ site.prodname }} with prometheus metrics enabled, it may be desirable to use network policy
+to limit access to {{ site.prodname }}'s metrics endpoints.
+
+### Purpose of this page
+
+Provide an example workflow for creating network policies to limit access to {{site.prodname}}'s metrics endpoints.
+
+## Prerequisites
+
+- {{site.prodname}} is installed with [prometheus metrics reporting enabled](TODO).
+- calicoctl is [installed in your PATH and configured to access the data store](TODO).
+
+## Overview
+
+The basic process is as follows:
+
+1. Create host endpoints for each node which you'd like to secure.
+1. Create a default network policy which allows traffic to and from the hosts.
+1. Create a network policy which denies unwanted traffic to the {{site.prodname}} metrics endpoints.
+1. Apply labels to allow access to the prometheus metrics.
+
+### Example
+
+1. List the nodes on which {{site.prodname}} is running. In this case, we have two nodes in the cluster.
+
+   ```shell
+   $ calicoctl get nodes
+   NAME
+   kubeadm-master
+   kubeadm-node-0
+   ```
+
+1. Create host endpoints for each {{site.prodname}} node.
+
+   Create a file named `host-endpoints.yaml` containing a host endpoint for each node listed
+   in step 1. In this example, the contents would look like this.
+
+   ```yaml
+   apiVersion: projectcalico.org/v3
+   kind: HostEndpoint
+   metadata:
+     name: kubeadm-master.eth0
+     labels:
+       running-calico-node: "true"
+   spec:
+     interfaceName: eth0
+     node: kubeadm-master
+     ports:
+     - name: prometheus
+       protocol: TCP
+       port: 9091
+   ---
+   apiVersion: projectcalico.org/v3
+   kind: HostEndpoint
+   metadata:
+     name: kubeadm-node-0.eth0
+     labels:
+       running-calico-node: "true"
+   spec:
+     interfaceName: eth0
+     node: kubeadm-node-0
+     ports:
+     - name: prometheus
+       protocol: TCP
+       port: 9091
+   ```
+
+   Note the use of a label to indicate that this host endpoint is running {{site.prodname}}, as well as
+   the named prometheus port. We'll use these to apply policy in a later step.
+
+   Then, use calicoctl to apply the host endpoints with the following command.
+
+   ```shell
+   calicoctl apply -f host-endpoints.yaml
+   ```
+
+1. Create a default network policy to allow traffic
```

Comment:
Creating host endpoints instantly gives default deny so we should recommend creating policy first (and explain why) since that will help people avoid gotchas.

Reply:
Yeah, I had wondered about this - creating the policy first is certainly better but as a walkthrough it's a less intuitive process.

I'll rework to do the policy first though, I think that's right.